Add `EFFECT_MUST_USE_SZONE`

Lyris12 opened the pull request:
For use with Link Spells:
> [Judgement Arrows](https://yugioh.fandom.com/wiki/Judgement_Arrows): `You can only control . . . "Judgement Arrows". . . in your Spell & Trap Zone a Link Monster points to.`
[Newtrix Nightveil](https://github.com/YGOCC/Moon/commit/8def1708bbf289c266d22c74b380f3507054ed28#diff-e36ce81fd83efc9bb7b027c59661d6f9): `Activate this card so it points to a monster you control.`

Original code by DailyShana, VanillaSalt, & edo9300

Lyris12 committed:
Add `EFFECT_MUST_USE_SZONE`

purerosefallen commented:
Please move the effect number to something else, rather than 1 more than the official effects.

Lyris12 committed:
Fix `EFFECT_MUST_USE_SZONE` as proposed by @purerosefallen

Change effect code
Add effect constant to `interpreter.cpp`
Now resets automatically
